### PR TITLE
[Enhancement] extends show data skew

### LINKED ISSUE #32575: [Enhancement] extends show data skew
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

SHOW DATA SKEW returns data distribution info. Now at least one partition must be specified, we need extend the command to show all partitions for a table, that is the command works for a table without a partition specified.

https://doris.apache.org/docs/dev/sql-manual/sql-reference/Show-Statements/SHOW-DATA-SKEW?_highlight=skew

### Solution

return contents like below.
PARTITION_NAME and original skew info.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xiedeyantu**
## Proposed changes

Issue Number: close #32575 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataSkewStmt.java [hunk 1]
```diff
@@ -58,9 +63,23 @@ public void analyze(Analyzer analyzer) throws UserException {
...
+        }
+
         PartitionNames partitionNames = tblRef.getPartitionNames();
-        if (partitionNames == null || partitionNames.getPartitionNames().size() != 1) {
-            throw new AnalysisException("Should specify one and only one partition");
+        if (partitionNames == null) {
+            if (tbl.get().getPartitionNames().isEmpty()) {
+                throw new AnalysisException("Can not find any partition");
+            }
+            tblRef.setPartitionNames(new PartitionNames(false, new ArrayList<>(tbl.get().getPartitionNames())));
```
Comment (rohitrs1983) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataSkewStmt.java:82]: do you think we should we also show tempPartitions? getPartitionNames() does not handle tempPartitions.

Comment (xiedeyantu) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataSkewStmt.java:82]: > do you think we should we also show tempPartitions? getPartitionNames() does not handle tempPartitions.

@rohitrs1983  OK, I will add both later.

**Comment by xiedeyantu** [comment]
run buildall

**Review by rohitrs1983 — APPROVED** [review]

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**